main.go

```diff
@@ -179,11 +180,15 @@ func main() {
 		}(i)
 	}
 
+	var shuttingDown bool
+
 	// Start one more goroutine for the HTTP server
 	httpServer := makeHTTPServer(aggregatedRedisClients, aggregatedMongoSessions, denylist, syncer)
 	go func() {
 		httpErr := httpServer.ListenAndServe()
-		if httpErr != nil {
+		if shuttingDown {
```

Comment:
is this a race condition?

Reply:
small deal since the only result is the wrong log message but worth making sure the handlers go off in the right order.

Reply:
Yes, but like you mentioned I'm not sure it matters.  The risk is that if it's shut down in the middle of an HTTP server error it won't get reported.  The handlers seem to work correctly from my testing.  We have three options here: 1) leave as-is, 2) include the HTTP error (messier), 3) add locks (more complexity).